### PR TITLE
Fix issue #10755 Canvas without contexts

### LINKED ISSUE #10755: Canvases without contexts should not draw anything
In `build_fragment_type_specific_display_items` we have code that deals with obtaining the pixels of a canvas element. In the event that the canvas has no context (ie. `canvas.getContext('2d')` hasn't been called) we currently synthesize a vector of opaque white pixels. This doesn't match what any other browser does, and we should just ignore it instead.

Code: `components/layout/display_list_builder.rs`
Test: a [reference](http://testthewebforward.org/docs/reftests.html) test needs to be [written](https://github.com/servo/servo/blob/master/tests/wpt/README.md#writing-new-tests) and added to `tests/wpt/mozilla/tests/css/`.

```html
<style>
div {
  width: 20px;
  height: 20px;
  background-color: green;
}
</style>
<div><canvas width="20" height="20"></canvas></div>
```

### PR DESCRIPTION
**Pull request by daoshengmu**
While canvas without contexts, we just need to return the function without drawing any pixel.

Fixes #10755.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10871)

### CODE REVIEW

File: tests/wpt/mozilla/tests/css/canvas_without_context.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<!doctype html>
+<html>
+<meta charset="utf-8">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
```
Comment (jdm) [inline tests/wpt/mozilla/tests/css/canvas_without_context.html]: This should be a [reference test](http://testthewebforward.org/docs/reftests.html) to verify that the rendered output matches what we expect.

**Comment by daoshengmu** [comment]
@jdm I have followed your suggestion to modify tests. Please have a look.

**Comment by jdm** [comment]
@daoshengmu That's better but we can improve the test a bit. The reference file should be one that doesn't use the feature being tested, so it should just be a div without any canvas at all. The test file should have the canvas without a context, since that's what we're trying to verify.

**Comment by jdm** [comment]
Oh, and it looks like there was an issue when rebasing, since there are some commits from other people included.

**Comment by jdm** [comment]
@bors-servo: r+
Thanks!

**Comment by daoshengmu** [comment]
@jdm. Thanks for your patience.

**Pull request merged**